modules/dispatcher/dispatcher.c

```diff
@@ -1281,22 +1283,40 @@ mi_response_t *ds_mi_set(const mi_params_t *params,
 	return init_mi_result_ok();
 }
 
-mi_response_t *ds_mi_list(int full)
+mi_response_t *ds_mi_list(const mi_params_t *params, int full)
 {
+	str partition_name;
+	int filter_partition = 0;
 	mi_response_t *resp;
 	mi_item_t *resp_obj;
 	mi_item_t *parts_arr, *part_item;
-	ds_partition_t *part_it;
+	ds_partition_t *partition, *part_it;
 
 	resp = init_mi_result_object(&resp_obj);
 	if (!resp)
 		return 0;
 
+	if (try_get_mi_string_param(params, "partition", &partition_name.s, &partition_name.len) == 0)
+		filter_partition = 1;
```

Comment:
I think you can get rid of the `filter_partition` flag and use `partition` for that purpose too. Init `partition` to NULL and have here the block for searching the partition by name, under this initial `if`.

Reply:
Thanks Bogdan, the recent push should address this.